File: OOPS_Concepts/Java/Encapsulation.java
/**
 * Encapsulation in Java
 * 
 * Encapsulation is the process of wrapping data (variables) and code (methods) together as a single unit.
 * In encapsulation, the variables of a class will be hidden from other classes, and can be accessed only through the methods of their current class.
 * Therefore, it is also known as data hiding.
 */

public class Person {
    private String name;
    private int age;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getAge() {
        return age;
    }

    public void setAge(int age) {
        this.age = age;
    }
}

public class Main {
    public static void main(String[] args) {
        Person person = new Person();
        person.setName("John");
        person.setAge(30);

        System.out.println("Name: " + person.getName() + ", Age: " + person.getAge());
    }
}

File: OOPS_Concepts/Python/Encapsulation.py
# Encapsulation in Python

class Person:
    def __init__(self, name, age):
        self.__name = name
        self.__age = age

    def get_name(self):
        return self.__name

    def set_name(self, name):
        self.__name = name

    def get_age(self):
        return self.__age

    def set_age(self, age):
        self.__age = age

person = Person("John", 30)
print(f"Name: {person.get_name()}, Age: {person.get_age()}")

person.set_name("Jane")
person.set_age(25)
print(f"Name: {person.get_name()}, Age: {person.get_age()}")
